File: java/src/main/java/com/experian/dvs/client/address/format/enrichment/Metadata.java
package com.experian.dvs.client.address.format.enrichment;

import com.experian.dvs.client.server.address.format.RestApiEnrichmentMetadata;

import java.util.Objects;

public class Metadata {

    private final String code;
    private final String message;
    private final String detail;


    public Metadata(RestApiEnrichmentMetadata api) {
        this.code = Objects.toString(api.getCode(), "");
        this.message = Objects.toString(api.getMessage(), "");
        this.detail = Objects.toString(api.getDetail(), "");
    }
    public String getCode() {
        return code;
    }

    public String getMessage() {
        return message;
    }

    public String getDetail() {
        return detail;
    }
}

File: csharp/DVSClient/DVSClient/address/format/enrichment/Metadata.cs
using DVSClient.Server.Address.Format;

namespace DVSClient.Address.Format.Enrichment
{
    public class Metadata
    {
        public string Code { get; }
        public string Message { get; }
        public string Detail { get; }

        public Metadata(RestApiEnrichmentMetadata api)
        {
            Code = api.Code ?? string.Empty;
            Message = api.Message ?? string.Empty;
            Detail = api.Detail ?? string.Empty;
        }
    }
}